service/ethevents.go

```diff
@@ -40,20 +41,37 @@ func EthEvents(
 		specs.Contracts,
 		specs.NetworkSource,
 		signer,
-		w3uri,
+		w3uris,
 		cm,
 		vocapp,
 		ethereumWhiteList,
 	)
+
 	if err != nil {
 		return fmt.Errorf("couldn't create ethereum events listener: %w", err)
 	}
 	for _, e := range evh {
 		ev.AddEventHandler(e)
 	}
-	go func() {
-		ev.SubscribeEthereumEventLogs(ctx, startBlock)
-	}()
+
+	initialized := new(bool)
```

Comment:
Why a pointer?

Reply:
no need, only one goroutine use the var, changed and no longer using a pointer